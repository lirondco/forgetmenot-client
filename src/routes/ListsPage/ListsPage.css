.ListsPage {
    padding: 5%;
}

.newlist_idea {
    display: flex;
    flex-direction: row;
}

.newlist_idea a {
    order: 1px solid #aaa;
    background-color: #FF6F00;
    font-size: 1.1em;
    border-radius: 4px;
    padding: 12px 12px;
    font-weight: bold;
    text-decoration: none;
    margin: 2%;
    flex: 6;
    text-align: center;
}