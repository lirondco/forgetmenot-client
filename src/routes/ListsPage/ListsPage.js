import React, { Component } from 'react'
import ListContext from '../../contexts/ListContext'
import IdeasApiService from '../../services/ideas-api-service'
import { Section } from '../../components/Utils/Utils'
import IdeaListItem from '../../components/IdeaListItem/IdeaListItem'
import { Link } from 'react-router-dom'
import './ListsPage.css'

export default class ListsPage extends Component {
  static defaultProps = {
    history: {
      push: () => {},
    },
  }
  static contextType = ListContext

  componentDidMount() {
    IdeasApiService.getLists()
      .then(this.context.setLists)
      .then(this.context.clearError)
      .catch(this.context.setError)
  }

  componentWillUnmount() {
    this.context.clearLists()
  }

  handleDeleteSuccess = list => {
    const { history } = this.props
    history.push(`/`)
    this.context.deleteList(list)
  }

  renderLists() {
    const { lists = [] } = this.context
    return lists.map(list =>
      <IdeaListItem 
        id={list.id}
        key={list.id}
        list={list}
        onDeleteSuccess={()=>this.handleDeleteSuccess(list.id)}
      />
    )
  }

  render() {
    const { error } = this.context
    return (
      <Section list className='ListsPage'>
        <div className='newlist_idea'>
         <Link to="/newidea">
            <div className="newidea">
                New Idea
            </div>
        </Link>          
         <Link to="/newlist">
            <div className="newList">
                New List
            </div>
        </Link>
        </div>
        {error
          ? <p className='red'>There was an error, try again</p>
          : this.renderLists()}
      </Section>
    )
  }
}