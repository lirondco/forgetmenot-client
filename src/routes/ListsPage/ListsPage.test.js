import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ListsPage from './ListsPage'

describe(`ListsPage component`, () => {
  it('renders a .ListsPage by default', () => {
    const wrapper = shallow(<ListsPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders an list in an IdeaListItem for each lists in array', () => {

    const context = {
      lists: [
        {
          "id": 1,
          "name": "List 1",
          "modified": "2019-01-03T00:00:00.000Z",
          "number_of_ideas": 12
        },
        {
          "id": 2,
          "name": "List 2",
          "modified": "2018-08-15T23:00:00.000Z",
          "number_of_ideas": 11
        },
        {
          "id": 3,
          "name": "List 3",
          "modified": "2018-03-01T00:00:00.000Z",
          "number_of_ideas": 0
        },
        {
          "id": 4,
          "name": "List 4",
          "modified": "2019-01-04T00:00:00.000Z",
          "number_of_ideas": 5
        },
      ]
    }
    const IdeasListItem = shallow(<ListsPage />, context)
      .find('IdeasListItem')
    expect(toJson(IdeasListItem)).toMatchSnapshot()
  })
})