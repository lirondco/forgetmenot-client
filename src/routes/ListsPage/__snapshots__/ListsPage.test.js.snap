// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListsPage component renders a .ListsPage by default 1`] = `
<Section
  className="ListsPage"
  list={true}
>
  <div
    className="newlist_idea"
  >
    <Link
      to="/newidea"
    >
      <div
        className="newidea"
      >
        New Idea
      </div>
    </Link>
    <Link
      to="/newlist"
    >
      <div
        className="newList"
      >
        New List
      </div>
    </Link>
  </div>
</Section>
`;

exports[`ListsPage component renders an list in an IdeaListItem for each lists in array 1`] = `null`;
